### ISSUE TITLE
ListIssueComments causes panic

### ISSUE THREAD

**Issue by hochhaus** [gogits/gogs#4587]
- Gogs version (or commit ref): 0.11.19
- Git version: git version 2.11.0
- Operating system: Debian Stretch
- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gogs.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist (usually found in `log/gogs.log`):

```
[Macaron] 2017-06-27 21:34:10: Started GET /api/v1/repos/[redacted]/sg/issues/94/comments for 127.0.0.1
[Macaron] PANIC: runtime error: invalid memory address or nil pointer dereference
/usr/local/go/src/runtime/panic.go:489 (0x4340bf)
/usr/local/go/src/runtime/panic.go:63 (0x432f6e)
/usr/local/go/src/runtime/signal_unix.go:290 (0x44930f)
/home/vagrant/gopath/src/github.com/gogits/gogs/models/repo.go:320 (0xb7aac6)
/home/vagrant/gopath/src/github.com/gogits/gogs/models/repo.go:333 (0xb7ac33)
/home/vagrant/gopath/src/github.com/gogits/gogs/models/repo.go:274 (0xb7a453)
/home/vagrant/gopath/src/github.com/gogits/gogs/models/repo.go:278 (0xb7a49f)
/home/vagrant/gopath/src/github.com/gogits/gogs/models/repo.go:282 (0xb7a54b)
/home/vagrant/gopath/src/github.com/gogits/gogs/models/issue.go:159 (0xb3c39b)
/home/vagrant/gopath/src/github.com/gogits/gogs/models/comment.go:138 (0xb335a3)
/home/vagrant/gopath/src/github.com/gogits/gogs/models/comment.go:146 (0xb336cf)
/home/vagrant/gopath/src/github.com/gogits/gogs/routers/api/v1/repo/issue_comment.go:36 (0xcd7077)
/usr/local/go/src/runtime/asm_amd64.s:514 (0x460608)
/usr/local/go/src/reflect/value.go:434 (0x4b279f)
/usr/local/go/src/reflect/value.go:302 (0x4b1d64)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:177 (0x88bd3f)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:137 (0x88b70a)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:121 (0x8a99e2)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:112 (0x8a9906)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/github.com/go-macaron/session/session.go:186 (0x901dc0)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:79 (0x8a9781)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:157 (0x88ba12)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:135 (0x88b80b)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:121 (0x8a99e2)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:112 (0x8a9906)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/recovery.go:161 (0x8bd26b)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/logger.go:40 (0x8ace98)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:157 (0x88ba12)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:135 (0x88b80b)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:121 (0x8a99e2)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:112 (0x8a9906)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/logger.go:52 (0x8bc4e4)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/logger.go:40 (0x8ace98)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:157 (0x88ba12)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/github.com/go-macaron/inject/inject.go:135 (0x88b80b)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/context.go:121 (0x8a99e2)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/router.go:187 (0x8be539)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/router.go:296 (0x8b65dd)
/home/vagrant/gopath/src/github.com/gogits/gogs/vendor/gopkg.in/macaron.v1/macaron.go:220 (0x8ae4dc)
/usr/local/go/src/net/http/server.go:2568 (0x6dc702)
/usr/local/go/src/net/http/server.go:1825 (0x6d8a12)
/usr/local/go/src/runtime/asm_amd64.s:2197 (0x462dc1)
[Macaron] 2017-06-27 21:34:10: Completed GET /api/v1/repos/[redacted]/sg/issues/94/comments 500 Internal Server Error in 23.578387ms
```

## Description

When trying to list all comments for an issue with the following command:

`curl -v -X GET -H "Authorization: token [redacted]" -H "Accept: application/json" -H "Content-Type: application/json" [redacted]/api/v1/repos/[redacted]/sg/issues/94/comments`

I receive the panic associated with the above stack trace. This might be caused because `loadAttributes()` in `models/comment.go` only populates a raw issue.

**Comment by hochhaus** [comment]
It appears that `loadAttributes()` cannot load the issue with `getIssueByID()` as it causes an infinite loop (with the issue loading comments and the comments loading the issues). Instead, only the `Repo` of the issue must be loaded:

```
ahochhaus@ahochhaus-pc:~/go/src/github.com/gogits/gogs$ git diff
diff --git a/models/comment.go b/models/comment.go
index be29e2ba..a9ec6efd 100644
--- a/models/comment.go
+++ b/models/comment.go
@@ -110,6 +110,12 @@ func (c *Comment) loadAttributes(e Engine) (err error) {
                if err != nil {
                        return fmt.Errorf("getIssueByID [%d]: %v", c.IssueID, err)
                }
+               if c.Issue.Repo == nil {
+                       c.Issue.Repo, err = getRepositoryByID(e, c.Issue.RepoID)
+                       if err != nil {
+                               return fmt.Errorf("getRepositoryByID [%d]: %v", c.Issue.RepoID, err)
+                       }
+               }
        }
 
        if c.Attachments == nil {
```

This isn't a clean solution but it works. Can you offer guidance on the correct way for me to fix this and I will send a PR?

**Comment by Unknwon** [comment]
This is claimed to be fixed by merging #4589, please help test on `develop` branch or https://try.gogs.io.

**Comment by hochhaus** [comment]
I built and tested from `develop` and can confirm that this fixes the issue. Thanks for the quick turn around.
